lib/strategy.js

```diff
@@ -223,7 +223,7 @@ OAuth2Strategy.prototype.authenticate = function(req, options) {
     }
 
     var state = options.state;
-    if (state) {
+    if (!state) {
```

Comment:
Should probably change to this
```
if (typeof state === 'undefined' || state === null)
```
so that we aren't relying on just truthy/falsy values